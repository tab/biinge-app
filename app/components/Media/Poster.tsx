import React from "react"
import { View } from "react-native"
import LinearGradient from "react-native-linear-gradient"
import { useTranslation } from "react-i18next"

import { posterStyles } from "styles"
import Image from "components/ui/Image"
import Rating from "components/Media/Rating"
import Badge from "components/Media/Badge"
import ContentRating from "components/Media/ContentRating"
import { Media } from "models/Media"

type Props = {
  item: Media
}

const PosterComponent = ({
  item: { image, contentRating, star, want, watched },
}: Props) => {
  const { t } = useTranslation()

  return (
    <View style={posterStyles.container}>
      <Image style={posterStyles.image} image={image} />
      <LinearGradient
        style={posterStyles.gradient}
        colors={[
          "rgba(0, 0, 0, 0) 100%)",
          "rgba(0, 0, 0, 0.4) 50%",
          "rgba(0, 0, 0, 0.6) 0%",
        ]}
      />
      <Rating>{star}</Rating>
      {(want || watched) && (
        <Badge>{want ? t("badge.want.title") : t("badge.watched.title")}</Badge>
      )}
      <ContentRating>{contentRating}</ContentRating>
    </View>
  )
}

export default PosterComponent
