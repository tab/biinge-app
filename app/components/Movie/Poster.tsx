import React from "react"
import { View, Image, Text } from "react-native"
import LinearGradient from "react-native-linear-gradient"

import { posterStyles, ratingStyles } from "styles"
import { RatingType } from "types"

type Props = {
  image: string
  rating: RatingType
}

const PosterComponent = ({ image, rating }: Props) => {
  return (
    <View style={posterStyles.container}>
      <Image
        style={posterStyles.image}
        resizeMode="cover"
        source={{ uri: image }}
      />
      <LinearGradient
        style={posterStyles.gradient}
        colors={[
          "rgba(0, 0, 0, 0) 100%)",
          "rgba(0, 0, 0, 0.4) 50%",
          "rgba(0, 0, 0, 0.6) 0%",
        ]}
      />
      <View style={ratingStyles.container}>
        <Text style={ratingStyles.rating}>{rating.star}</Text>
      </View>
    </View>
  )
}

export default PosterComponent
