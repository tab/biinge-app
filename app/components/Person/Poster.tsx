import React from "react"
import { View } from "react-native"
import Animated, { FadeIn } from "react-native-reanimated"
import LinearGradient from "react-native-linear-gradient"
import { BlurView } from "@react-native-community/blur"
import { useTheme } from "@react-navigation/native"

import { APP_APPEARANCE_DARK, APP_APPEARANCE_LIGHT } from "config"
import { formatDate } from "helpers/formatDate"
import Image from "components/ui/BWImage"
import Title from "components/ui/Title"
import Typography from "components/ui/Typography"
import { personPosterStyles, personStyles } from "styles"
import colors from "styles/colors"

type Props = {
  poster_path: string
  name: string
  birthday: string
}

const PosterComponent = ({ poster_path, name, birthday }: Props) => {
  const { dark } = useTheme()

  return (
    <View style={personPosterStyles.root}>
      <BlurView
        style={personPosterStyles.blur}
        blurType={dark ? APP_APPEARANCE_DARK : APP_APPEARANCE_LIGHT}
        blurAmount={15}
        reducedTransparencyFallbackColor={colors.white}
      />
      <View style={personPosterStyles.card}>
        <Image
          style={personPosterStyles.image}
          size="w780"
          path={poster_path}
        />
        <LinearGradient
          style={personPosterStyles.blur}
          start={{ x: 1, y: 0 }}
          end={{ x: 1, y: 1 }}
          colors={[
            "rgba(0,0,0,0)",
            "rgba(0,0,0,0)",
            "rgba(0,0,0,0.5)",
            "rgba(0,0,0,1)",
          ]}
        />
        <Animated.View
          style={personPosterStyles.content}
          entering={FadeIn.delay(500)}
        >
          <Title style={personStyles.title}>{name}</Title>
          {birthday && (
            <Typography variant="headline" style={personStyles.date}>
              {formatDate(birthday)}
            </Typography>
          )}
        </Animated.View>
      </View>
    </View>
  )
}

export default PosterComponent
