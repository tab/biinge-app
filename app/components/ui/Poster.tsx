import React from "react"
import { View } from "react-native"
import { BlurView } from "@react-native-community/blur"
import { useTheme } from "@react-navigation/native"

import { APP_APPEARANCE_DARK, APP_APPEARANCE_LIGHT } from "config"
import { posterStyles } from "styles"
import Image from "components/ui/Image"
import colors from "styles/colors"

type Props = {
  poster_path: string
}

const PosterComponent = ({ poster_path }: Props) => {
  const { dark } = useTheme()

  return (
    <>
      <BlurView
        style={[
          posterStyles.blur,
          dark ? posterStyles.blurDark : posterStyles.blurLight,
        ]}
        blurType={dark ? APP_APPEARANCE_DARK : APP_APPEARANCE_LIGHT}
        blurAmount={15}
        reducedTransparencyFallbackColor={colors.white}
      />
      <View style={posterStyles.root}>
        <Image style={posterStyles.image} size="w780" path={poster_path} />
      </View>
    </>
  )
}

export default PosterComponent
