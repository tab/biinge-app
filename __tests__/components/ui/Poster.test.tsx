import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"
import { it } from "@jest/globals"

import Poster from "components/ui/Poster"

describe("Poster", () => {
  const imgPath = "eO3r38fwnhb58M1YgcjQBd3VNcp.jpg"

  it("renders correctly", () => {
    const tree = render(<Poster posterPath={imgPath} />)
    expect(tree).toMatchSnapshot()
  })
})
