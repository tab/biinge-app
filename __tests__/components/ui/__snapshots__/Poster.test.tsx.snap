// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster renders correctly 1`] = `
[
  <BlurView
    blurAmount={15}
    blurType="light"
    reducedTransparencyFallbackColor="#ffffff"
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        {
          "backgroundColor": "rgba(255, 255, 255, 0.7)",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />,
  <View
    style={
      {
        "paddingBottom": 24,
        "paddingHorizontal": 36,
        "paddingVertical": 24,
      }
    }
  >
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      entering={[Function]}
      style={
        {
          "alignSelf": "center",
          "aspectRatio": "4/6",
          "backgroundColor": "#d1cecf",
          "borderRadius": 12,
          "elevation": 3,
          "shadowColor": "#181818",
          "shadowOffset": {
            "height": 5,
            "width": 2,
          },
          "shadowOpacity": 0.7,
          "shadowRadius": 25,
          "width": 525,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 16,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
        />
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "alignSelf": "center",
              "aspectRatio": "4/6",
              "backgroundColor": "#d1cecf",
              "borderRadius": 12,
              "elevation": 3,
              "shadowColor": "#181818",
              "shadowOffset": {
                "height": 5,
                "width": 2,
              },
              "shadowOpacity": 0.7,
              "shadowRadius": 25,
              "width": 525,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          source={
            {
              "priority": "normal",
              "uri": "https://image.tmdb.org/t/p/w780eO3r38fwnhb58M1YgcjQBd3VNcp.jpg",
            }
          }
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>,
]
`;
